### ISSUE TITLE
Refresh_Token Expires within a few hours

### ISSUE THREAD

**Issue by Khungersumit** [FortnoxAB/csharp-api-sdk#234]
Hi 

I see some issues with refresh_token.

After getting authCode, I got access_token(valid for 1 hr) and refresh_token(valid for 31 days, I saved in DB)
After 1 hr access_token is expired and I generated a new access_token using the refresh_token(from my DB).
Now I got new access_token and refresh_token (I updated the new refresh_token in my DB)

After a few hours(>7-8 hours) API says "**_Request failed: Invalid refresh token_**"  

That is not the expected behavior.
Has anyone faced a similar issue before? :)

**Comment by xpagedeveloper** [comment]
I haven’t seen any similar issue. I suggest you create some kind of audit log with the tokens. 
And create an issue with fortnox support.

**Comment by RadhikeCalcey** [comment]
@Khungersumit Any updates on this?

**Comment by Mmmattias** [comment]
Same issue here.

Created the token in the afternoon and the morning after it was invalid.

I guess this is not related to the C# client library but still a vital issue to solve if we are to be able to work with our integration.

**Comment by richardrandak** [comment]
Did you generated the tokens with the Fortnox SDK?
If not, make sure to add `access_type=offline` parameter to the authorization URL.. not doing that could result in short-lived token.. but that's just guessing. I have not encountered this issue myself. 

Just so you know, this repository is not affiliated with Fortnox anymore. If you feel the issue is not caused by this library, please write directly to Fortnox official support and describe the problem to them.

**Comment by yasirjanjua** [comment]
@xpagedeveloper I am making blocking calls to refresh the tokens but the token gets invalidated on higher request rates. I sent 30 requests simultaneously and the error was reproducible.

**Comment by xpagedeveloper** [comment]
1). How often do you generate access and refresh tokens?
1 -5 time per day and customer

2). What is the expected/average number of Apis calls per customer (for which you generate the tokens) on an average day you make?

aprox 1-30 /customer and day

**Comment by kkrogulski** [comment]
If you login with the same Fortnox API user again, you invalidate every previous refresh token.

**Comment by xpagedeveloper** [comment]
> If you login with the same Fortnox API user again, you invalidate the previous refresh token.

Yes if you request a new access token using a refresh token any previous refresh tokens is invalid.
If you authorize the same application again that will make any previous refresh tokens invalid.
If you login using the web with the user that validated the API access that should not invalidate any refreshtokens.
